### PR TITLE
Restore the proper scale of Lab and Lch

### LINKED ISSUE #49: Lab and Lch values
Continuing discussion from #48.

Lab values are currently scaled, L by 100 and a and b by 128. Even though a and b are unbound, typical ranges are -128 to 128. 

By my limited research so far, I have not seen scaled lab values used ( unlike rgb which is usually assumed in the range 0-1 ). For example Conversion formula's, Color Difference formula's and most research papers assume unscaled lab values (I am happy to be corrected on this). 

Right now using the lab spaces implies unnecessary scaling when used practically. So I am proposing that the Lab values should be changed to be unscaled. Same for Lch values.

If the decision is to still scale the value, we should provide a new_from_unscaled and into_unscaled function, to get values in and out.

### PR DESCRIPTION
**Pull request by sidred**
This restores the scale of Lab and Lch to be the same as in all of the specifications within reach. That is:

- Lab::l and Lch::l goes from 0 to 100,
- Lab::a and Lab::b goes from -128 to 127,
- Lch::chroma goes from 0 to about 128-181.

Also added additional tests for Lab <-> Lch and Lab<->Xyz conversions. Removed tests for lab and lch from the color mine data as it is not accurate.

This is a breaking change, since everything that uses these values directly will be wrong. Closes #49.

### CODE REVIEW

**Comment by sidred** [comment]
I have also added tests from the color checker data http://www.babelcolor.com/colorchecker-2.htm

This has additional data for RgbSpaces to Cie tests. Once the rgb standards are added, these can be used as a comprehensive test for gamma encoded color space to cie tests.

File: src/lab.rs [hunk 1]
```diff
@@ -401,22 +401,22 @@ mod test {
     #[test]
     fn red() {
         let a = Lab::from(Rgb::new(1.0, 0.0, 0.0));
-        let b = Lab::new(53.23288 / 100.0, 80.09246 / 128.0, 67.2031 / 128.0);
-        assert_relative_eq!(a, b, epsilon = 0.0001);
+        let b = Lab::new(53.23288, 80.09246, 67.2031);
+        assert_relative_eq!(a, b, epsilon = 0.01);
```
Comment (Ogeon) [inline src/lab.rs]: Does the test precision need to be this low?

**Comment by Ogeon** [comment]
This is great, and look at those decimals! I think we are set for life with that precision, given that it's also correct. I'm sure it is.

File: src/lab.rs (hunk 1 above)
Comment (sidred) [inline src/lab.rs]: I don't know the source of this data and this does not pass with 0.001. Anyway the Lab - Rgb should be taken care of in the color checker data, which is more accurate.

**Comment by sidred** [comment]
Yeah, but I was hoping it would be within the default epsilon :)

File: src/lab.rs (hunk 1 above)
Comment (Ogeon) [inline src/lab.rs]: I seem to remember that I recalculated those numbers in my attempt at this, but we can leave it as it is. It's still checked in other tests, as you said.

**Comment by Ogeon** [comment]
> `epsilon = 0.000000000001`

:heart_eyes:

Sorry, it's just so refreshing after the blunt xyY numbers from color mine. Are we ready to go?

**Comment by sidred** [comment]
Yes

**Comment by Ogeon** [comment]
Great! @homu r+

**Pull request merged**